doc/zephyr.doxyfile.in

```diff
@@ -1996,7 +1996,8 @@ PREDEFINED             = "CONFIG_ARCH_HAS_CUSTOM_BUSY_WAIT" \
                          "__aligned(x)=" \
                          "__printf_like(x, y)=" \
                          "__attribute__(x)=" \
-                         "__syscall="
+                         "__syscall=" \
+                         "DT_DOXYGEN"
```

Comment:
unrelated, undocumented, change?  What's this for?  (Is this commit from some other PR?  @mbolivar-nordic)

Later: Looks like it's used to generate documentation in Doxygen, but IMO this should not be named to suggest it's restricted to devicetree usage.  There are many other cases where conent that is conditionally included in a header should be included or excluded for Doxygen-generated material.  In other projects I've done this in `doxygen.cfg.in`:
```
# Use "#if defined(@PROJECT_UC@_DOXYGEN)" when providing a stub declaration
# to which documentation is attached.
# Use "/* @cond DOXYGEN_EXCLUDE */ ... /* @endcond */" around the real
# declarations.
PREDEFINED += "@PROJECT_UC@_DOXYGEN"
```

Reply:
This was pull in from PR #23891 to get the commits from that PR related to devicetree/zephyr.h

Reply:
OK.  Now I see this:
```
 * Getting doxygen to play along with all the dts-specific ifdeffery
￼ * proved too complex for DT_CHOSEN_ZEPHYR_FLASH_LABEL, so we document
￼ * everything under a DT_DOXYGEN define.
```
in the point of use, so I guess it's explained.  I still think a generic "only emit this when running under Doxygen" solution is more useful, but that's probably more hassle to get through.